server/src/handlers/influxdb.rs

```diff
@@ -71,6 +77,303 @@ impl From<Bytes> for WriteRequest {
 
 pub type WriteResponse = ();
 
+/// Influxql response organized in the same way with influxdb
+#[derive(Debug, Serialize)]
+pub struct InfluxqlResponse {
+    pub results: Vec<InfluxqlResult>,
+}
+
+#[derive(Debug)]
+pub struct InfluxqlResult {
+    statement_id: u32,
+    series: Option<Vec<OneSeries>>,
+}
+
+impl Serialize for InfluxqlResult {
+    fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let mut influxql_result = serializer.serialize_map(Some(4))?;
+        influxql_result.serialize_entry("statement_id", &self.statement_id)?;
+        if let Some(series) = &self.series {
+            influxql_result.serialize_entry("series", series)?;
+        }
+
+        influxql_result.end()
+    }
+}
+
+#[derive(Debug)]
+struct OneSeries {
+    name: String,
+    tags: Option<Tags>,
+    columns: Vec<String>,
+    values: Vec<Vec<Datum>>,
+}
+
+impl Serialize for OneSeries {
+    fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let mut one_series = serializer.serialize_map(Some(4))?;
+        one_series.serialize_entry("name", &self.name)?;
+        if let Some(tags) = &self.tags {
+            one_series.serialize_entry("tags", &tags)?;
+        }
+        one_series.serialize_entry("columns", &self.columns)?;
+        one_series.serialize_entry("values", &self.values)?;
+
+        one_series.end()
+    }
+}
+
+#[derive(Debug)]
+struct Tags(Vec<(String, String)>);
+
+impl Serialize for Tags {
+    fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let mut tags = serializer.serialize_map(Some(self.0.len()))?;
+
+        for (tagk, tagv) in &self.0 {
+            tags.serialize_entry(tagk, tagv)?;
+        }
+
+        tags.end()
+    }
+}
+
+/// [InfluxqlResult] builder
+#[derive(Default)]
+pub struct InfluxqlResultBuilder {
+    /// Id of the related influxql
+    statement_id: u32,
+
+    /// Schema of influxql query result
+    ///
+    /// Its format is like:
```

Comment:
Give one example and link
```bash
{"results":[{"statement_id":0,"series":[{"name":"mymeas",
                                         "columns":["time","myfield","mytag1","mytag2"],
                                         "values":[["2017-03-01T00:16:18Z",33.1,null,null],
                                                   ["2017-03-01T00:17:18Z",12.4,"12","14"]]}]}]}


```
- https://docs.influxdata.com/influxdb/v1.8/tools/api/#query-data-with-a-select-statement

The above result is for query without aggregator
```
$ curl -XPOST 'http://localhost:8086/query?db=mydb' --data-urlencode 'q=SELECT * INTO "newmeas" FROM "mymeas"'
````

And the format is different when there are aggregator in query.
```bash
{"results":[{"statement_id":0,"series":[{"name":"mymeas",
                                         "columns":["time","myfield"],
                                         "tags": {"os": "Ubuntu16.04LTS"},
                                         "values":[["2017-03-01T00:16:18Z",33.1],
                                                   ["2017-03-01T00:17:18Z",12.4]]}]}]}

```
- `columns` only contains time and fields
- `tags` is add to display tags

Reply:
`tags` just will be added when defining some in `group by` with or without aggregator as I test before implementing.
+ no group by + aggregator
```
curl --get "http://localhost:8086/query?org=antgroup&bucket=influxql-test2" \
  --header "Authorization: Token kvjuMDtW7gZbnXQVGEwNJd2sUUBx5baT5tK89_4zMUwMV3WDcXksT8bIcjTW1LXDH364kMU1khwcx8cNVBv12g==" \
  --data-urlencode "db=influxql-test2" \
  --data-urlencode "rp=autogen" \
  --data-urlencode "q=SELECT COUNT(temp) FROM home"
{"results":[{"statement_id":0,"series":[{"name":"home","columns":["time","count"],"values":[["1970-01-01T00:00:00Z",26]]}]}]}
```
+ group by + aggregator
```
curl --get "http://localhost:8086/query?org=antgroup&bucket=influxql-test2" \
  --header "Authorization: Token kvjuMDtW7gZbnXQVGEwNJd2sUUBx5baT5tK89_4zMUwMV3WDcXksT8bIcjTW1LXDH364kMU1khwcx8cNVBv12g==" \
  --data-urlencode "db=influxql-test2" \
  --data-urlencode "rp=autogen" \
  --data-urlencode "q=SELECT COUNT(temp) FROM home group by room"
{"results":[{"statement_id":0,"series":[{"name":"home","tags":{"room":"Kitchen"},"columns":["time","count"],"values":[["1970-01-01T00:00:00Z",13]]},{"name":"home","tags":{"room":"Living Room"},"columns":["time","count"],"values":[["1970-01-01T00:00:00Z",13]]}]}]}
```
+ no group by + no aggregator(has shown in above)
+ group by + no aggregator
```
{"results":[{"statement_id":0,"series":[{"name":"home","tags":{"room":"Kitchen"},"columns":["time","co","hum","temp"],"values":[["2022-01-01T08:00:00Z",0,35.9,21],["2022-01-01T09:00:00Z",0,36.2,23],["2022-01-01T10:00:00Z",0,36.1,22.7],["2022-01-01T11:00:00Z",0,36,22.4],["2022-01-01T12:00:00Z",0,36,22.5],["2022-01-01T13:00:00Z",1,36.5,22.8],["2022-01-01T14:00:00Z",1,36.3,22.8],["2022-01-01T15:00:00Z",3,36.2,22.7],["2022-01-01T16:00:00Z",7,36,22.4],["2022-01-01T17:00:00Z",9,36,22.7],["2022-01-01T18:00:00Z",18,36.9,23.3],["2022-01-01T19:00:00Z",22,36.6,23.1],["2022-01-01T20:00:00Z",26,36.5,22.7]]},{"name":"home","tags":{"room":"Living Room"},"columns":["time","co","hum","temp"],"values":[["2022-01-01T08:00:00Z",0,35.9,21.1],["2022-01-01T09:00:00Z",0,35.9,21.4],["2022-01-01T10:00:00Z",0,36,21.8],["2022-01-01T11:00:00Z",0,36,22.2],["2022-01-01T12:00:00Z",0,35.9,22.2],["2022-01-01T13:00:00Z",0,36,22.4],["2022-01-01T14:00:00Z",0,36.1,22.3],["2022-01-01T15:00:00Z",1,36.1,22.3],["2022-01-01T16:00:00Z",4,36,22.4],["2022-01-01T17:00:00Z",5,35.9,22.6],["2022-01-01T18:00:00Z",9,36.2,22.8],["2022-01-01T19:00:00Z",14,36.3,22.5],["2022-01-01T20:00:00Z",17,36.4,22.2]]}]}]}
```

Reply:
I think link may be enough?
It is a the format in influxdb, maybe it is unnecessary to explain it in ceresdb?